frame/democracy/src/lib.rs

```diff
@@ -579,21 +578,13 @@ decl_module! {
 
 		/// Cancel a proposal queued for enactment.
 		#[weight = SimpleDispatchInfo::FixedOperational(10_000)]
-		fn cancel_queued(
-			origin,
-			#[compact] when: T::BlockNumber,
-			#[compact] which: u32,
-			#[compact] what: ReferendumIndex
-		) {
+		fn cancel_queued(origin, which: ReferendumIndex) {
 			ensure_root(origin)?;
-			let which = which as usize;
-			let mut items = <DispatchQueue<T>>::get(when);
-			if items.get(which).and_then(Option::as_ref).map_or(false, |x| x.1 == what) {
-				items[which] = None;
-				<DispatchQueue<T>>::insert(when, items);
-			} else {
-				Err("proposal not found")?
-			}
+			let mut items = <DispatchQueue<T>>::get();
+			let original_len = items.len();
+			items.retain(|i| i.2 != which);
```

Comment:
Does it matter that this function will continue to check items even if a match is found?

Something ever so slightly improved would stop after the first match since `Index` is unique.

Reply:
there will only ever be one of them (the same RefIndex cannot be dispatched twice), and it's `O(N)` in both instances. i think the simpler expression is more readable.